tests/proto3/proto3_compat_test.go

```diff
@@ -108,6 +108,49 @@ func TestFixedU64Roundtrip(t *testing.T) {
 	assert.EqualValues(t, p3ToAm.Int64, amToP3.Int64)
 }
 
+func TestMultidimensionalSlices(t *testing.T) {
+	s := [][]int8{
+		[]int8{1, 2},
+		[]int8{3, 4, 5}}
+
+	_, err := cdc.MarshalBinaryBare(s)
+	assert.Error(t, err, "expected error: multidimensional slices are not allowed")
```

Comment:
What about a test with more than 2 dimensions? And what about tests that mix arrays and slices (!)?